kivy/core/text/markup.py

```diff
@@ -165,7 +165,7 @@ def _pre_render(self):
         # flattened later when shortening and broken up lines if broken
         # mid-word will have space mid-word when lines are joined
         uw_temp = None if shorten else uw
-        xpad = options['padding_x']
+        xpad = options['padding'][0]
```

Comment:
Since we also have another case at lines just below, that uses `xpad` for justifying text, that needs to be checked, is probably better to just use `options['padding'][0]` and `options['padding'][2]` instead of a generic `xpad` variable.
 (Would be nice to have some test that exercises this case)